tls/s2n_resume.h

```diff
@@ -99,3 +104,13 @@ extern int s2n_store_to_cache(struct s2n_connection *conn);
  * once we release the session resumption API. */
 int s2n_config_set_initial_ticket_count(struct s2n_config *config, uint8_t num);
 int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t num);
+
+struct s2n_session_ticket {
+    struct s2n_blob ticket_data;
+    uint32_t session_lifetime;
+};
+
+typedef int (*s2n_session_ticket_cb)(struct s2n_connection *conn, struct s2n_session_ticket *ticket);
+int s2n_config_set_session_ticket_callback(struct s2n_config *config, s2n_session_ticket_cb cb);
+int s2n_session_ticket_get_data(struct s2n_session_ticket *ticket, uint8_t **data, size_t *data_len);
```

Comment:
In s2n_session_ticket_get_data, we're passing a pointer to our memory to the customer. Do you think we should be accepting a pointer to customer memory and then copying the ticket to it instead? It might be a little confusing that the data returned here has the same lifetime as the session ticket, but I'm not sure which model we prefer.

Reply:
I'm not super concerned which we do *provided* that it is clearly documented in all cases. (Ideally, we should be consistent, but that might not always be possible.) Related to that, should the type of `data` be `const uint8_t **`? I always get confused when you combine consts with pointers. Ideally, we should be indicating that while address contained in `data` may change, the information referred to by the address must remain constant.